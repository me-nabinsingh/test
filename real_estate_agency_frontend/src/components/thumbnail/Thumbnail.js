import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './styles.css';

class  Thumbnail extends Component{

    render() {
        const { images } = this.props;
        if(images.length === 0) {
            return null;
        }
        const url = images[0].url;
        
        var divStyle = {
            backgroundImage: 'url(' + url + ')'
        }

        return ( 
            <div className="thumbnail" style={divStyle}>
                &nbsp;
            </div>
        );
    }
}

Thumbnail.defaultProps = {
    images: []
}

Thumbnail.propTypes = {
    images: PropTypes.array
}
export default Thumbnail;