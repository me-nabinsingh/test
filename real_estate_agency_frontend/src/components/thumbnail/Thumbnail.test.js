import React from 'react';
import Thumbnail from './Thumbnail';
import renderer from 'react-test-renderer';

const images = [{
    url: "https://s3.us-east-2.amazonaws.com/nod.realestateagency/15442744174651200px-Mercury_in_color_-_Prockter07-edit1.jpg"
}];

describe('<Thumbnail />', () => {
    it("doesn't break if images object is not passed", () => {
        

        const component = renderer.create(<Thumbnail />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        
        const component = renderer.create(<Thumbnail images={images} />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});