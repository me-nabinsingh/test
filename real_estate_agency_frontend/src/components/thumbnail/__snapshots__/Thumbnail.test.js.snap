// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail /> doesn't break if images object is not passed 1`] = `null`;

exports[`<Thumbnail /> renders without crashing 1`] = `
<div
  className="thumbnail"
  style={
    Object {
      "backgroundImage": "url(https://s3.us-east-2.amazonaws.com/nod.realestateagency/15442744174651200px-Mercury_in_color_-_Prockter07-edit1.jpg)",
    }
  }
>
  Â 
</div>
`;
